sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/aio/_client.py

```diff
@@ -268,16 +273,22 @@ def key_id(self) -> str:
             key = result.key
 
         """
-
-        result = await self._client.unwrap_key(
-            self._key_id.vault_endpoint,
-            self._key_id.name,
-            self._key_id.version,
-            algorithm=algorithm,
-            value=encrypted_key,
-            **kwargs
-        )
-        return UnwrapResult(key_id=self.key_id, algorithm=algorithm, key=result.result)
+        local_key = await self._get_local_key(**kwargs)
+        if local_key and local_key.is_private_key():
+            if "unwrapKey" not in self._allowed_ops:
```

Comment:
Should these be enums or some other globally defined representation?

Reply:
Sure, that's arguably more stylish. Changing them isn't urgent and would be a great first contribution for someone, so I opened #8138.